caluma/form/historical_schema.py

```diff
@@ -190,13 +190,13 @@ def resolve_value(self, info, as_of, *args):
         answerdocuments.sort(key=lambda x: x.sort)
 
         documents = [
-            models.Document.history.filter(id=ad.document_id).filter(
-                history_date__lte=as_of
-            )[0]
-            for ad in answerdocuments
+            models.Document.history.filter(
+                id=ad.document_id, history_date__lte=as_of
+            ).latest("history_date")
+            for ad in list(dict.fromkeys(answerdocuments))
```

Comment:
`AnswerDocument`s and `Document`s can be dict keys? Didn't know!

I'd like a comment to explain how/why this works though. Might confuse future readers

Reply:
Done.

Reply:
Awesome! Just for future reference: https://docs.python.org/3.6/whatsnew/3.6.html#whatsnew36-compactdict